pywbem_mock/_cimrepository.py

```diff
@@ -883,67 +912,77 @@ def EnumerateClassNames(self, **params):
         # Validate namespace and get class_store for this namespace
         class_store = self.get_class_store(namespace)
 
-        classname = params.get('ClassName', None)
-        if classname:
-            assert(isinstance(classname, CIMClassName))
-            if not class_store.exists(classname.classname):
+        if ClassName:
+            assert(isinstance(ClassName, CIMClassName))
+            if not class_store.exists(ClassName.classname):
                 raise CIMError(
                     CIM_ERR_INVALID_CLASS,
                     _format("The class {0!A} defined by 'ClassName' parameter "
                             "does not exist in namespace {1!A}",
-                            classname, namespace))
+                            ClassName, namespace))
 
-        clns = self._get_subclass_names(classname, class_store,
-                                        params['DeepInheritance'])
+        clns = self._get_subclass_names(ClassName, class_store,
+                                        DeepInheritance)
         rtn_clns = [
             CIMClassName(cn, namespace=namespace, host=self.host)
             for cn in clns]
 
-        return self._make_tuple(rtn_clns)
+        return rtn_clns
 
     ####################################################################
     #
     #   GetClass server request
     #
     ####################################################################
 
-    def GetClass(self, **params):
+    def GetClass(self, ClassName, namespace=None, LocalOnly=None,
+                 IncludeQualifiers=None, IncludeClassOrigin=None,
+                 PropertyList=None):
         """
-        Implements a mock server responder for
+        Implements a cim repository responder  for
         :meth:`~pywbem.WBEMConnection.GetClass
 
         Retrieve a CIM class from the local repository.
 
         For a description of the parameters, see
         :meth:`pywbem.WBEMConnection.GetClass`.
+
+        Returns:
+            :class:`~pywbem.CIMClass`
+
+        Raises:
+            CIMError: CIM_ERR_INVALID_NAMESPACE: invalid namespace,
+            CIMError: CIM_ERR_INVALID_CLASS: class defined by the ClassName
+            parameter does not exist in the CIM repository
         """
         namespace = self.namespace
```

Comment:
You disabled `namespace = self.namespace` for EnumerateClasses. I think it should be disabled here as well.